### DISCUSSION TITLE
Querying Flowise API from python doesn't use buffer memory?

### DISCUSSION THREAD

**Discussion by Hyperlooper** [FlowiseAI/Flowise#1376] [Q&A]
Hello, I have a simple chain with buffer memory piped in to the conversational agent. This works properly using the embedded chat window in the webpage, but when I query the same chain via python using the API, the agent does not have any memory message to message. How do you query the agent via the API in a way that does preserve this memory state?

Working via UI
![image](https://github.com/FlowiseAI/Flowise/assets/7346880/64e86c0a-88a9-4026-8dfc-62b0d0134c24)

Does not work via this call: 

> def queryFlowise(payload):
>     response = requests.post(API_URL, json=payload)
>     print(response)
>     return response.json()

**Comment by HenryHengZJ** [comment]
@Hyperlooper 
for buffer memory, you have to supply the list of chat messages as `history` in the API body - https://docs.flowiseai.com/integrations/memory/short-term-memory

alternative is to use memory nodes with sessionId like Redis, Zep, Upstash etc, and supply the `sessionId` in the body:
https://docs.flowiseai.com/integrations/memory/long-term-memory

**Reply by VincentEmmanuel** [reply]
can you elaborate on how to use buffer memory in the API?
